src/v/kafka/client/test/reconnect.cc

```diff
@@ -74,3 +76,81 @@ FIXTURE_TEST(reconnect, kafka_client_fixture) {
     info("Stopping client");
     client.stop().get();
 }
+inline http::client make_admin_client() {
+    net::base_transport::configuration transport_cfg;
+    transport_cfg.server_addr = net::unresolved_address{"127.0.0.1", 9644};
+    return http::client(transport_cfg);
+}
+
+FIXTURE_TEST(password_change_live_client, kafka_client_fixture) {
+    ss::sstring username{"admin"};
+    ss::sstring userpass{"foopar"};
+
+    info("Waiting for leadership");
+    wait_for_controller_leadership().get();
+
+    info("Enable SASL and restart");
+    enable_sasl_and_restart(username);
+
+    ss::sstring user_body = fmt::format(
+      "\"username\": \"{}\", \"password\": \"{}\", \"algorithm\": "
+      "\"SCRAM-SHA-256\"",
+      username,
+      userpass);
+    user_body
+      = "{" + user_body
+        + "}"; // Curly braces here because they are delim symbols to fmt
```

Comment:
nit, it might be easier to read like:
```suggestion
    ss::sstring user_body = fmt::format(
      R"("{{username": "{}", "password": "{}",\"algorithm": "SCRAM-SHA-256"}})",
      username,
      userpass);
```

Reply:
I swear I tried that and I swear it caused problems because of the curly braces and fmt. Sigh, I'll do it again. You just have the magic touch.